db.go

```diff
@@ -127,48 +163,95 @@ func (x *dummyAuthenticator) ResetPasswordFinish(identity string, token string,
 	}
 }
 
-func (x *dummyAuthenticator) CreateIdentity(identity, password string) error {
+func (x *dummyUserStore) CreateIdentity(identity, password string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
-	if _, exists := x.users[CanonicalizeIdentity(identity)]; !exists {
-		x.users[CanonicalizeIdentity(identity)] = &dummyUser{identity, password, ""}
+	user := getUser(x.users, identity)
+	if user == nil {
+		userId := generateUserId(x.users)
+		x.users[userId] = &dummyUser{userId, identity, password, ""}
```

Comment:
You should canonicalize the identity before doing anything with it. In particular, here you are storing a non-canonicalized identity.